src/App/Support/Traits/Wc_Order_Settings_Trait.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tamara_Checkout\App\Support\Traits;
+
+trait Wc_Order_Settings_Trait {
+
+	public function get_default_billing_country_code(): string {
+		return ! empty( $this->get_currency_to_country_mapping()[ get_woocommerce_currency() ] )
+			? $this->get_currency_to_country_mapping()[ get_woocommerce_currency() ]
+			: $this->get_store_base_country_code();
+	}
+
+	/**
+	 * Get country code based on its currency
+	 *
+	 * @return array
+	 */
+	public function get_currency_to_country_mapping(): array {
+		return [
+			'SAR' => 'SA',
+			'AED' => 'AE',
+			'KWD' => 'KW',
+			'BHD' => 'BH',
+		];
+	}
+
+	/**
+	 * Get store base country code
+	 *
+	 * @return string
+	 */
+	public function get_store_base_country_code(): string {
```

Comment:
Consider to move this to a Helper as well, there's a method in Tamara_Widget, you may need to refactor it as well